keps/sig-apps/1847-autoremove-statefulset-pvcs/README.md

```diff
@@ -0,0 +1,453 @@
+
+
+### Scalability
+
+* **Will enabling / using this feature result in any new API calls?**
+
+  Yes and no. This feature will result in additional resource deletion calls, which will
+  scale like the number of pods in the stateful set (ie, one PVC per pod and possibly one
+  PV per PVC depending on the reclaim policy). There will not be additional watches,
+  because the existing pod watches will be used (note: this is to be confirmed in the
+  code).
+  
+  However, anyone who uses this feature would have made those resource deletions anyway:
+  those PVs cost money. So overall there shouldn't be much increase beyond the
+  second-order effect of this feature allowing more automation.
+
+* **Will enabling / using this feature result in introducing new API types?**
+  No.
+
+* **Will enabling / using this feature result in any new calls to the cloud 
+provider?**
+  PVC deletion may cause PV deletion, depending on reclaim policy, which will result in
+  cloud provider calls through the volume API. However, as noted above, these calls would
+  have been happening anyway, manually.
+
+* **Will enabling / using this feature result in increasing size or count of 
+the existing API objects?**
+  - PVC, new ownerRef.
+
+* **Will enabling / using this feature result in increasing time taken by any 
+operations covered by existing SLIs/SLOs?**
+  No. (are there StatefulSet SLOs?)
+
+* **Will enabling / using this feature result in non-negligible increase of 
+resource usage (CPU, RAM, disk, IO, ...) in any components?**
+  No.
+
+### Troubleshooting
+
+_TBD on beta graduation._
+
+The Troubleshooting section currently serves the `Playbook` role. We may consider
+splitting it into a dedicated `Playbook` document (potentially with some monitoring
+details). For now, we leave it here.
+
+* **How does this feature react if the API server and/or etcd is unavailable?**
+
+* **What are other known failure modes?**
+  For each of them, fill in the following information by copying the below template:
+  - [Failure mode brief description]
+    - Detection: How can it be detected via metrics? Stated another way:
+      how can an operator troubleshoot without logging into a master or worker node?
+    - Mitigations: What can be done to stop the bleeding, especially for already
+      running user workloads?
+    - Diagnostics: What are the useful log messages and their required logging
+      levels that could help debug the issue?
+      Not required until feature graduated to beta.
+    - Testing: Are there any tests for failure mode? If not, describe why.
+
+* **What steps should be taken if SLOs are not being met to determine the problem?**
+
+[supported limits]: https://git.k8s.io/community//sig-scalability/configs-and-limits/thresholds.md
+[existing SLIs/SLOs]: https://git.k8s.io/community/sig-scalability/slos/slos.md#kubernetes-slisslos
+
+## Implementation History
+
+  - KEP created.
```

Comment:
can you add a milestone here to date this? (ie 1.21)

Reply:
done